ostelco-ios-client/Network/Analytics.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Analytics.swift
+//  ostelco-ios-client
+//
+//  Created by mac on 4/29/19.
+//  Copyright © 2019 mac. All rights reserved.
+//
+
+import Foundation
+import ostelco_core
+import FirebaseAnalytics
+
+class ROAnalytics {
+    static func logEvent(_ event: AnalyticsEvent) {
+        Analytics.logEvent(event.name, parameters: event.metadata)
+    }
+}
+
+enum AnalyticsEvent {
+    case LegalStuffAgreed
+    case EnteredNickname
+    case ChosenCountry(country: Country)
+    case ChosenIDMethod(idMethod: String)
+    case DownloadingESIM
+}
+
+extension AnalyticsEvent {
+    var name: String {
+        switch self {
+        case .ChosenCountry:
+            return "chosen_country"
+        case .ChosenIDMethod:
+            return "chosen_id_method"
+        default:
+            return String(describing: self).snakeCased()! // TODO: Is this safe enough to force unwrap?
+        }
+    }
+    
+    var metadata: [String : String] {
+        switch self {
+        case .ChosenCountry(let country):
+            return ["country": country.countryCode]
+        case .ChosenIDMethod(let idMethod):
+            return ["id_method": idMethod]
+        default:
+            return [:]
+        }
+    }
+}
+
+// Found it on the internet: https://gist.github.com/dmsl1805/ad9a14b127d0409cf9621dc13d237457
```

Comment:
You also could [steal `JSONEncoder`'s implementation](https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/JSONEncoder.swift#L135) from the stdlib

Reply:
@designatednerd By stealing, you mean copy / pasting that version inside the string extension?

Reply:
Basically. I would double check the licenses we're using vs. they're using before doing that vs. re-implementing something similar, but my recollection is the stdlib license is quite permissive.

Reply:
This has been pulled into an extension so we can deal with that as we see fit later.